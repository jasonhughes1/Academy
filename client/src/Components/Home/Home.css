
.home-container {
  background-color: #3C4040;
  display: flex;
  justify-content: center;
  height: 100vh;
}

.link-container {
  display: flex;
  flex-wrap: wrap;
  background-color: white;
  border: 4px solid #48a9b6;
  flex-direction: column;
  width: 60%;
}

.description {
  font-size: 30px;
  margin-top: 15px;
  width: 85%;
}

.button-area {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin: 0px;
  padding: 15px 0px;
}

.home-nav {
  color: black;
  font-size: 20px;
}

.charts-nav {
  color: black;
  font-size: 20px;
}

.charts-nav {
  margin-top: 10px;
  color: black;
}

.home-nav:hover {
  border-color: transparent;
  box-shadow: 0 0 0 2px #48a9b6;
}

.charts-nav:hover {
  border-color: transparent;
  box-shadow: 0 0 0 2px #48a9b6;
}
