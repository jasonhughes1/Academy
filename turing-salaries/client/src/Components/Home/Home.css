
.home-container {
  background-color: #3C4040;
  height: 100vh;
}

.link-container {
  display: flex;
  justify-content: space-between;
}
