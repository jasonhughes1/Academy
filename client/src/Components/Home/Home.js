import React, { Component } from 'react';
import './Home.css';
import { NavLink } from 'react-router-dom';
import Instafeed from 'react-instafeed';


class Home extends Component {
  render() {
      const instafeedTarget = 'instafeed';
    return (
      <div className='home-container'>
        <div className='link-container'>
          <span className="description">
            Welcome to Turing Salaries where you
            can view data from past Turing graduates.
          </span>
          <div className="button-area">
            <NavLink className='home-nav'
              to='/viewsalary'>Browse Salaries</NavLink>
            <NavLink className='charts-nav'
              to='/viewline'>2015 & 2016 Salaries</NavLink>
            <NavLink className='charts-nav'
              to='/viewPie'>Jobs Titles of Grads</NavLink>
            <NavLink className='charts-nav'
              to='/viewBar'>How many days to hire?</NavLink>
          </div>
        </div>
        <h2>Recent Instagram Posts</h2>
        <div class='insta-container'
           id={instafeedTarget}>
          <Instafeed
            limit='10'
            ref='instafeed'
            resolution='standard_resolution'
            sortBy='most-recent'
            target={instafeedTarget}
            template=''
            userId='295628134'
            clientId='clientIdInstagramApiString'
            accessToken='295628134.17babc8.7801cc7efb6d462dacfbf4a3222c992b'
          />
        </div>
      </div>
    );
  }
}

export default Home;
