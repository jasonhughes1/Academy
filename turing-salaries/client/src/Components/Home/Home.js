import React, { Component } from 'react';
import './Home.css';

class Home extends Component {
  render() {
    return (
      <div>
        <div className='image-container'>    
        </div>
      </div>
    )
  }
}

export default Home;
