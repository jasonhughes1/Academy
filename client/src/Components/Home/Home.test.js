import React from 'react';
import { shallow } from 'enzyme';
import Home from './Home';

describe('Home Tests', () => {

  it('should be defined', () => {
    const renderedHome = shallow(<Home />);

    expect(renderedHome).toBeDefined();
  });

  it('should match the snap shot', () => {
    const renderedHome = shallow(<Home />);

    expect(renderedHome).toMatchSnapshot();
  });

});
