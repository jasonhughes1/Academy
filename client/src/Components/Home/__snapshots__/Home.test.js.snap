// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Tests should match the snap shot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="link-container"
>
        <span
                className="description"
        >
                Welcome to Turing Salaries where you can view data from past Turing graduates.
        </span>
        <div
                className="button-area"
        >
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="home-nav"
                        to="/viewsalary"
                >
                        Browse Salaries
                </NavLink>
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="charts-nav"
                        to="/viewline"
                >
                        2015 & 2016 Salaries
                </NavLink>
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="charts-nav"
                        to="/viewPie"
                >
                        Jobs Titles of Grads
                </NavLink>
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="charts-nav"
                        to="/viewBar"
                >
                        How many days to hire?
                </NavLink>
        </div>
</div>,
      "className": "home-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="description"
>
            Welcome to Turing Salaries where you can view data from past Turing graduates.
</span>,
          <div
            className="button-area"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="home-nav"
                        to="/viewsalary"
            >
                        Browse Salaries
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="charts-nav"
                        to="/viewline"
            >
                        2015 & 2016 Salaries
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="charts-nav"
                        to="/viewPie"
            >
                        Jobs Titles of Grads
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="charts-nav"
                        to="/viewBar"
            >
                        How many days to hire?
            </NavLink>
</div>,
        ],
        "className": "link-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to Turing Salaries where you can view data from past Turing graduates.",
            "className": "description",
          },
          "ref": null,
          "rendered": "Welcome to Turing Salaries where you can view data from past Turing graduates.",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="home-nav"
                to="/viewsalary"
>
                Browse Salaries
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="charts-nav"
                to="/viewline"
>
                2015 & 2016 Salaries
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="charts-nav"
                to="/viewPie"
>
                Jobs Titles of Grads
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="charts-nav"
                to="/viewBar"
>
                How many days to hire?
</NavLink>,
            ],
            "className": "button-area",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Browse Salaries",
                "className": "home-nav",
                "to": "/viewsalary",
              },
              "ref": null,
              "rendered": "Browse Salaries",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "2015 & 2016 Salaries",
                "className": "charts-nav",
                "to": "/viewline",
              },
              "ref": null,
              "rendered": "2015 & 2016 Salaries",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Jobs Titles of Grads",
                "className": "charts-nav",
                "to": "/viewPie",
              },
              "ref": null,
              "rendered": "Jobs Titles of Grads",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "How many days to hire?",
                "className": "charts-nav",
                "to": "/viewBar",
              },
              "ref": null,
              "rendered": "How many days to hire?",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="link-container"
>
          <span
                    className="description"
          >
                    Welcome to Turing Salaries where you can view data from past Turing graduates.
          </span>
          <div
                    className="button-area"
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="home-nav"
                              to="/viewsalary"
                    >
                              Browse Salaries
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="charts-nav"
                              to="/viewline"
                    >
                              2015 & 2016 Salaries
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="charts-nav"
                              to="/viewPie"
                    >
                              Jobs Titles of Grads
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="charts-nav"
                              to="/viewBar"
                    >
                              How many days to hire?
                    </NavLink>
          </div>
</div>,
        "className": "home-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="description"
>
              Welcome to Turing Salaries where you can view data from past Turing graduates.
</span>,
            <div
              className="button-area"
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="home-nav"
                            to="/viewsalary"
              >
                            Browse Salaries
              </NavLink>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="charts-nav"
                            to="/viewline"
              >
                            2015 & 2016 Salaries
              </NavLink>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="charts-nav"
                            to="/viewPie"
              >
                            Jobs Titles of Grads
              </NavLink>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="charts-nav"
                            to="/viewBar"
              >
                            How many days to hire?
              </NavLink>
</div>,
          ],
          "className": "link-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Turing Salaries where you can view data from past Turing graduates.",
              "className": "description",
            },
            "ref": null,
            "rendered": "Welcome to Turing Salaries where you can view data from past Turing graduates.",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="home-nav"
                  to="/viewsalary"
>
                  Browse Salaries
</NavLink>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="charts-nav"
                  to="/viewline"
>
                  2015 & 2016 Salaries
</NavLink>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="charts-nav"
                  to="/viewPie"
>
                  Jobs Titles of Grads
</NavLink>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="charts-nav"
                  to="/viewBar"
>
                  How many days to hire?
</NavLink>,
              ],
              "className": "button-area",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Browse Salaries",
                  "className": "home-nav",
                  "to": "/viewsalary",
                },
                "ref": null,
                "rendered": "Browse Salaries",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "2015 & 2016 Salaries",
                  "className": "charts-nav",
                  "to": "/viewline",
                },
                "ref": null,
                "rendered": "2015 & 2016 Salaries",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Jobs Titles of Grads",
                  "className": "charts-nav",
                  "to": "/viewPie",
                },
                "ref": null,
                "rendered": "Jobs Titles of Grads",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "How many days to hire?",
                  "className": "charts-nav",
                  "to": "/viewBar",
                },
                "ref": null,
                "rendered": "How many days to hire?",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
